addons/survey/wizard/survey_invite.py

```diff
@@ -170,12 +170,23 @@ def _prepare_answers(self, partners, emails):
             if self.existing_mode == 'resend':
                 partners_done = existing_answers.mapped('partner_id')
                 emails_done = existing_answers.mapped('email')
-                answers |= existing_answers
+
+                # only add the last answer for each user of each type (partner_id & email)
+                # to have only one mail sent per user
+                for partner_done in partners_done:
```

Comment:
I would avoid filtered because of prefetch nightmare it would do it there is a lot of existing answers X lot of partners_done . List comprehension should be sufficient: next([a for a in answers if a).sorted()) .

Reply:
Good point. Should be better now (if I understood correctly).